governator-core/src/main/java/com/netflix/governator/internal/PreDestroyLifecycleActions.java

```diff
@@ -2,55 +2,103 @@
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 import javax.annotation.PreDestroy;
 
 import com.netflix.governator.LifecycleAction;
 
 /**
- * Special AbstractLifecycleFeature to support @PreDestroy annotation processing.
- * Note that this feature is implicit in LifecycleModule and therefore does not need 
- * to be added using the LifecycleFeature multibinding.
+ * Special AbstractLifecycleFeature to support @PreDestroy annotation processing and
+ * java.io.AutoCloseable detection. Note that this feature is implicit in LifecycleModule 
+ * and therefore does not need to be added using the LifecycleFeature multibinding.
  * 
  * @author elandau
  */
-public final class PreDestroyLifecycleActions extends AbstractLifecycleFeature {
+public final class PreDestroyLifecycleActions extends AbstractLifecycleFeature<Set<String>> {
```

Comment:
AbstractLifecycleFeature has 4 introspection methods of any consequence, and the new 'context' argument needs to be a consistently-typed parameter passed into each one.  Using a generic signature for the context type means no casting for the subclasses, and flexibility for other implementations to use java.util.Map or whatever type works best for them (possibly null).  Coordination of argument types across methods seems legitimate use of generics; is there a down side?